src/components/Article.jsx

```diff
@@ -1,9 +1,39 @@
-import React from 'react'
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { fetchArticleById } from "../api";
+import { Card, CardContent} from "@mui/material";
 
 function Article() {
+  const [article, setArticle] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+  const { article_id } = useParams();
+
+  useEffect(() => {
+    setIsLoading(true);
+    fetchArticleById(article_id)
+      .then((data) => {
+        setArticle(data);
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        setError("Failed to load the article");
+        setIsLoading(false);
+      });
+  }, [article_id]);
```

Comment:
You don't need the article_id here. This will only change by navigating to a new page, which as you have a useEffect will cause a re render anyway. You do need to have the array, but you don't need anything inside it.